### PR TITLE
Remove deprecated API

### PR DESCRIPTION
**Pull request by 1chandu**
- Remove everything that was deprecated in v3 and v4.
- Add details to `upgrade-guide.md`
- Summarize v5 changes in `whats-new.md`

Verified with `test-browser` and `examples`.

### CODE REVIEW

File: docs/api-reference/webgl/framebuffer.md [hunk 1]
```diff
@@ -68,7 +68,6 @@ model.draw({
 
 Binding a framebuffer for multiple render calls
 ```js
-import {withState} from 'luma.gl';
```
Comment (ibgreen) [inline docs/api-reference/webgl/framebuffer.md]: Change to `import {withParameters}`?

File: docs/upgrade-guide.md [hunk 2]
```diff
@@ -1,5 +1,86 @@
 # Upgrade Guide
 
+## Upgrading from v4 to v5
+
+Please read this documentation before upgrading your luma.gl's dependency from v4 to v5. In v5 a number of previously deprecated features have been removed and a number of additional deprecations have been made at the same time.
```
Comment (ibgreen) [inline docs/upgrade-guide.md]: Remove the `s`: ` luma.gl's` ->  `luma.gl`

File: docs/upgrade-guide.md [hunk 3]
```diff
@@ -1,5 +1,86 @@
...
+  Model(gl)
+  Model(gl, {...opts});
+  Model(gl, {program});
+```
+
+Following style construction was deprecated in v4 and is now removed in v5.
+
+```
+  // NOT SUPPORTED
+  Model(gl)
```
Comment (ibgreen) [inline docs/upgrade-guide.md]: These are the same as above!

File: docs/upgrade-guide.md [hunk 4]
```diff
@@ -1,5 +1,86 @@
...
+| ---                  | ---              | --      |
+| `withState`          | `withParameters` | State management |
+| `glContextWithState` | `withParameters` | State management |
+|`withParameters({frameBuffer})`| `withParameters({framebuffer})`| State management |
+| `MONOLITHIC_SHADERS` | `MODULAR_SHADERS` | default shaders |
+| `Camera`, `PerspectiveCamera`, `OrthoCamera` | |
+| `Scene` | `None` | |
+| `isWebGLContext` | `isWebGL` | WebGL context validation |
+| `isWebGL2Context` | `isWebGL2` | WebGL2 context validation |
+| `generateMipmaps` | `mipmaps` | `Texture` constructor argument |
```
Comment (ibgreen) [inline docs/upgrade-guide.md]: I'd make a separate table for Texture class changes. The rest are global classes, functions.

File: docs/upgrade-guide.md [hunk 5]
```diff
@@ -1,5 +1,86 @@
...
+   positions: new Float32Array([ ... ]),
+   colors: {
+     size: 4,
+     value: new Float32Array([ ... ])
+   }
+ }
+});
+```
+
+### Removed Features
```
Comment (ibgreen) [inline docs/upgrade-guide.md]: Have luma.gl math classes been removed?

File: src/index.js [hunk 6]
```diff
@@ -151,12 +151,7 @@ export { // Moved to math.gl
   SphericalCoordinates
 } from 'math.gl';
```
Comment (ibgreen) [inline src/index.js]: At least we don't need to export experimental math.gl classes like SphericalCoordinates.

File: docs/api-reference/webgl/framebuffer.md (hunk 1 above)
Comment (1chandu) [reply docs/api-reference/webgl/framebuffer.md]: actually `withState` was not used in this example.

File: docs/upgrade-guide.md (hunk 5 above)
Comment (1chandu) [reply docs/upgrade-guide.md]: I removed ` src/deprecated/math-old.js`, are you referring to that?

File: docs/upgrade-guide.md (hunk 4 above)
Comment (1chandu) [reply docs/upgrade-guide.md]: sure will do.

**Pull request merged**